pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleLoadManagerImpl.java

```diff
@@ -702,7 +702,7 @@ private synchronized void doLoadRanking() {
                 }
 
                 // update metrics
-                if (resourceUnit.getResourceId().contains(hostname)) {
+                if (resourceUnit.getResourceId().startsWith(hostname + ":")) {
```

Comment:
nit: is ":" required?

Reply:
A contains match doesn't make sense. Let's say if you have "host1", "host10" and "host100". The resource id is host:port so that's why ":" is added and startsWith is used

Reply:
It might be possible that this change should be left in a separate PR. It just hurts my eyes to see fuzzy match using contains. 😆